pages/availability/troubleshoot.tsx

```diff
@@ -29,108 +27,92 @@ export default function Troubleshoot({ user }: inferSSRProps<typeof getServerSid
-      .then((res) => {
-        return res.json();
-      })
-      .then((availableIntervals) => {
-        setAvailability(availableIntervals.busy);
-        setLoading(false);
-      })
-      .catch((e) => {
-        console.error(e);
-      });
-  };
-
   useEffect(() => {
+    const fetchAvailability = (date: Dayjs) => {
+      const dateFrom = date.startOf("day").utc().format();
+      const dateTo = date.endOf("day").utc().format();
+      setLoading(true);
+
+      fetch(`/api/availability/${user.username}?dateFrom=${dateFrom}&dateTo=${dateTo}`)
+        .then((res) => {
+          return res.json();
+        })
+        .then((availableIntervals) => {
+          setAvailability(availableIntervals.busy);
+        })
+        .catch((e) => {
+          console.error(e);
+        })
+        .finally(() => {
+          setLoading(false);
+        });
+    };
     fetchAvailability(selectedDate);
   }, [selectedDate]);
 
-  if (loading) {
-    return <Loader />;
-  }
-
   return (
-    <div>
-      <Shell heading={t("troubleshoot")} subtitle={t("troubleshoot_description")}>
-        <div className="bg-white max-w-xl overflow-hidden shadow rounded-sm">
-          <div className="px-4 py-5 sm:p-6">
-            {t("overview_of_day")}{" "}
-            <input
-              type="date"
-              className="inline border-none h-8 p-0"
-              defaultValue={selectedDate.format("YYYY-MM-DD")}
-              onBlur={(e) => {
-                setSelectedDate(dayjs(e.target.value));
-              }}
-            />
-            <small className="block text-neutral-400">{t("hover_over_bold_times_tip")}</small>
-            <div className="mt-4 space-y-4">
-              <div className="bg-black overflow-hidden rounded-sm">
-                <div className="px-4 sm:px-6 py-2 text-white">
-                  {t("your_day_starts_at")} {convertMinsToHrsMins(user.startTime)}
-                </div>
-              </div>
-              {availability.map((slot) => (
-                <div key={slot.start} className="bg-neutral-100 overflow-hidden rounded-sm">
-                  <div className="px-4 py-5 sm:p-6 text-black">
-                    {t("calendar_shows_busy_between")}{" "}
-                    <span className="font-medium text-neutral-800" title={slot.start}>
-                      {dayjs(slot.start).format("HH:mm")}
-                    </span>{" "}
-                    {t("and")}{" "}
-                    <span className="font-medium text-neutral-800" title={slot.end}>
-                      {dayjs(slot.end).format("HH:mm")}
-                    </span>{" "}
-                    {t("on")} {dayjs(slot.start).format("D")}{" "}
-                    {t(dayjs(slot.start).format("MMMM").toLowerCase())} {dayjs(slot.start).format("YYYY")}
-                  </div>
-                </div>
-              ))}
-              {availability.length === 0 && <Loader />}
```

Comment:
This was causing an infinite loading since the call was successful but the busy slots were zero.